### DISCUSSION TITLE
"gatsby-plugin-sitemap" Cannot read property 'length' of undefined

### DISCUSSION THREAD

**Discussion by xbaha** [gatsbyjs/gatsby#31334] [Help]
Hello,
I copied the exact code here: https://github.com/gatsbyjs/gatsby/discussions/22757
but when I tried to add options to sitemap plugin, I got this:

```
Cannot read property 'length' of undefined

  64 |             }
  65 |
> 66 |             reporter.verbose(_internals.REPORTER_PREFIX + " Filtering " + allPages.length + " pages     based on " + excludes.length + " excludes");
```

not sure where can I start, I have no clue what the error means.

regards

**Comment by mwskwong** [comment]
That's a one-year-old code and there has been a major version update with breaking changes since then. You should follow the official documentations first.
